2-mandatory/3-financial-times.js

```diff
@@ -1,40 +1,72 @@
 /*
     Imagine you are working on the Financial Times web site! They have a list of article titles stored in an array.
-
     The home page of the web site has a headline section, which only has space for article titles which are 65 characters or less.
     Implement the function below, which will return a new array containing only article titles which will fit.
 */
-function potentialHeadlines(allArticleTitles) {
-    // TODO
+
+function potentialHeadlines(allArticleTitles)
+{
+    let AcceptedArticles = []
+
+    for (let i = 0; i < allArticleTitles.length; i++)
+    {
+        if (allArticleTitles[i].length <= 65)
+        {
+            AcceptedArticles.push(allArticleTitles[i]);
+        }
+    }
+    return AcceptedArticles;
 }
 
 /*
     The editor of the FT likes short headlines with only a few words!
     Implement the function below, which returns the title with the fewest words.
     (you can assume words will always be seperated by a space)
 */
-function titleWithFewestWords(allArticleTitles) {
-    // TODO
+
+function titleWithFewestWords(allArticleTitles)
+{
+    const shorter = (left, right) => left.length <= right.length ? left : right;
+
+    return allArticleTitles.reduce(shorter);
 }
 
```

Comment:
Could you please add some comments to this function, I didn't really understand it?